gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/TagService_FindIdsByAndReferenceTest.java

```diff
@@ -0,0 +1,122 @@
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TagService_FindIdsByAndReferenceTest {
+
+    private static final String ORG_ID = "DEFAULT";
+
+    @InjectMocks
+    private TagServiceImpl tagService = new TagServiceImpl();
+
+    @Mock
+    private TagRepository tagRepository;
+
+    @Mock
+    private ApiTagService apiTagService;
+
+    @Mock
+    private AuditService auditService;
+
+    @Mock
+    private GroupService groupService;
+
+    @BeforeEach
+    public void init() {
+        reset(tagRepository);
+    }
+
+    @Test
+    public void should_not_throw_exception_if_empty_tag_ids_parameter() throws TechnicalException {
+        when(tagRepository.findByIdsAndReference(new HashSet<>(), ORG_ID, TagReferenceType.ORGANIZATION)).thenReturn(Set.of());
+        try {
+            tagService.checkTagsExist(new HashSet<>(), ORG_ID, io.gravitee.rest.api.model.TagReferenceType.ORGANIZATION);
+        } catch (Exception e) {
+            fail("Should not throw exception");
+        }
+    }
+
+    @Test
+    public void should_throw_error_if_no_tags_found() throws TechnicalException {
+        when(tagRepository.findByIdsAndReference(Set.of("tag-1", "tag-2"), ORG_ID, TagReferenceType.ORGANIZATION)).thenReturn(Set.of());
+
+        assertThrows(TagNotFoundException.class, () -> tagService.checkTagsExist(Set.of("tag-1", "tag-2"), ORG_ID, io.gravitee.rest.api.model.TagReferenceType.ORGANIZATION));
+
+        verify(tagRepository, times(1)).findByIdsAndReference(any(), any(), any());
+    }
+
+    @Test
+    public void should_throw_error_if_only_one_tag_found() throws TechnicalException {
+        Tag tag1 = getRepositoryTag("tag-1");
+        Set<String> tags = new HashSet<>();
+        tags.add("tag-1");
+        tags.add("tag-2");
+
+        when(tagRepository.findByIdsAndReference(tags, ORG_ID, TagReferenceType.ORGANIZATION)).thenReturn(Set.of(tag1));
+
+        assertThrows(TagNotFoundException.class, () -> tagService.checkTagsExist(tags, ORG_ID, io.gravitee.rest.api.model.TagReferenceType.ORGANIZATION));
+
+        verify(tagRepository, times(1)).findByIdsAndReference(any(), any(), any());
+    }
+
+    @Test
+    public void should_return_tag_entities_when_all_tags_valid() throws TechnicalException {
+        Tag tag1 = getRepositoryTag("tag-1");
+        Tag tag2 = getRepositoryTag("tag-2");
+
+        when(tagRepository.findByIdsAndReference(Set.of("tag-1", "tag-2"), ORG_ID, TagReferenceType.ORGANIZATION)).thenReturn(Set.of(tag1, tag2));
+
+        try {
+            tagService.checkTagsExist(Set.of("tag-1", "tag-2"), ORG_ID, io.gravitee.rest.api.model.TagReferenceType.ORGANIZATION);
+        } catch (Exception e) {
+            fail("Should not throw exception");
+        }
```

Comment:
The try/catch is unnecessary, if an exception is thrown JUnit will catch it and flag the test as failed